teerex/src/tests/test_cases.rs

```diff
@@ -70,12 +76,45 @@ fn add_and_remove_dcap_enclave_works() {
 			signer.clone()
 		));
 		assert!(!<SovereignEnclaves<Test>>::contains_key(&signer));
-		assert_eq!(list_enclaves(), vec![])
+		assert_eq!(list_sovereign_enclaves(), vec![])
+	})
+}
+
+#[test]
+fn add_and_remove_dcap_proxied_enclave_works() {
+	new_test_ext().execute_with(|| {
+		Timestamp::set_timestamp(TEST_VALID_COLLATERAL_TIMESTAMP);
+
+		let alice = AccountKeyring::Alice.to_account_id();
+		register_test_quoting_enclave::<Test>(alice.clone());
+		register_test_tcb_info::<Test>(alice.clone());
+
+		let instance_address = EnclaveInstanceAddress {
+			fingerprint: TEST1_DCAP_QUOTE_MRENCLAVE.into(),
+			registrar: alice.clone(),
+			signer: AnySigner::try_from(TEST1_DCAP_QUOTE_SIGNER).unwrap(),
+		};
+
+		assert_ok!(Teerex::register_sgx_enclave(
+			RuntimeOrigin::signed(alice.clone()),
+			TEST1_DCAP_QUOTE.to_vec(),
+			None,
+			SgxAttestationMethod::Dcap { proxied: true }
+		));
+		assert_eq!(list_proxied_enclaves().len(), 1);
+		assert!(<ProxiedEnclaves<Test>>::contains_key(&instance_address));
```

Comment:
I have rerun the tests, and can confirm that without our added padding it does not work, and I don't see anything suspicious here.